components/searchable_channel_list.jsx

```diff
@@ -19,6 +19,14 @@ import {t} from 'utils/i18n';
 const NEXT_BUTTON_TIMEOUT_MILLISECONDS = 500;
 
 export default class SearchableChannelList extends React.Component {
+    static getDerivedStateFromProps(props) {
+        if (!props.isSearch) {
```

Comment:
This needs to compare the previous value of `props.isSearch` against the previous one or else random prop updates might send the user back to the first page of results